Bebb-Protocol-and-Apps/PWS
# Add webportals:  Enhanced Webportals Integration and UI Improvements in A-Frame Scene

This Pull Request introduces a series of changes that significantly improve the usage and visualization of webportals in the A-Frame scene. These modifications aim to provide an interactive and more engaging user experience in the VR environment.

Key Changes Include:

3D Webportals: Space Neighbors can now be visualized in the A-Frame scene for the Space. If not in VR/fullscreen mode, the Webportal shows an iframe to the Neighbor. If in VR/fullscreen mode, the Webportal shows a default image. The Webportal can be clicked to go to the Neighbor (its URL).

Dynamic Menu Addition: A new feature has been added that allows the dynamic addition of an in-VR menu. This menu is attached to the user's camera view, always available to interact, regardless of the viewer's orientation in the VR environment.

Interactive Menu: The VR menu includes a selection of actions, toggling between an "Open Menu" and "Close Menu" state. When opened, users have access to additional buttons or actions such as "Load Neighbors".

Aesthetic Enhancements: The UI for the menu has been redesigned to deliver a professional and pleasant visual experience. This includes adaptive size adjustments, and other minor design tweaks.

Webportals in VR/Fullscreen Mode: A code check has been added to determine whether the user is in VR/Fullscreen mode, thus managing the visibility of webportals in these different views for better interaction.

Please review the changes, and feel free to provide your feedback or suggest any improvements.

To test:
3D Webportals: Load a Space and use the VR menu in the bottom-left corner. Click on Load Neighbors. If the Space doesn't have any Neighbors yet, a message will appear for 4 seconds. If the Space has Neighbors (with valid URLs), they will be visualized further back in the scene as door-like elements along one line. If clicked, you should be taken to the Neighbor (Website).

Dynamic Menu Addition: After loading the A-Frame scene, a new menu should be visible in the camera view. Test if the menu follows the camera movement and remains visible regardless of the viewer's orientation in the VR environment.

Interactive Menu: Interact with the menu. It should respond to clicks and toggle between an "Open Menu" and "Close Menu" state. When opened, additional buttons or actions, such as "Load Neighbors", should appear.

Aesthetic Enhancements: Assess the visual appeal and functionality of the new menu. The menu should adjust its size depending on the state (opened or closed).

Webportals in VR/Fullscreen Mode: Enter VR/Fullscreen mode and check how the webportals behave. They should be appropriately managed based on the mode, with a focus on improving user interaction.
